src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +886,81 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+//
+// We are selecting the track in the following manner.
+// We divide the library tracks into two section . For which
+// we sort the library according to times_played and select a
+// percentage_of_prefered_tracks (25% by default) .
+//
+// What we do then is either select a random track from this %25
+// or a random track from the remaining 75% both with equal
+// probability .
+//
+// (Assuming that 25% tracks are unplayed or very rarely played)
+//
+// This ensures that there is a 50% chance that an rarely played
+// track is returned .
+//
+// This is better than returning the least played track because
+// in that case there may be instances where the same track is
+// returned by this function on multiple successive calls which is
+// not desired.
+//
+// Further this also does not restrict our function to only retrieve
+// unplayed tracks (there is probably a reason they are unplayed).
+///
+int AutoDJCratesDAO::getRandomTrackIdFromLibrary(void) {
```

Comment:
I hope the intent here is clear . Further this can be easiy tweaked to vary the probability to whatever seems logically optimum . 
For the current value of 25% , and say 100 tracks . The least played 25 has a probability of being selected of   
````
1/2*1/25 = 1/50
````
 and the remaining relatively frequently played tracks have a probability of getting selected of 
````
1/2 * 1/75 = 1/150 .
````
That is there is 3times more chance that a relatively unplayed track is selected. Suggestions ?